Support prepending of base url in markdown image paths

<!--
  Consider requesting the feature on https://docusaurus-2.netlify.com/feedback/ instead.
  Feature requests on v1 are more likely to be missed as we're focusing on building v2.
-->

## 🚀 Feature

I want to display images without using JSX in my Markdown code

```
![my image](/static/img/...)
```

and have Docusaurus modify the image `src` using the project's baseUrl.

### Have you read the [Contributing Guidelines on issues]

Yes

## Motivation & Pitch

Currently, it's recommended to link to an image using:

```markdown
import useBaseUrl from '@docusaurus/useBaseUrl'

<img alt="ORY Oathkeeper with Prometheus and Grafana" src={useBaseUrl('img/docs/grafana.png')} />
```

This has several disadvantages. First it doesn't work with native markdown and thus looks confusing when browsing the docs on e.g. GitHub (both in preview and raw view!):

<img width="1136" alt="Bildschirmfoto 2020-05-07 um 11 12 49" src="https://user-images.githubusercontent.com/3372410/81276757-c03b0480-9053-11ea-93c2-e48a9523a724.png">

<img width="1077" alt="Bildschirmfoto 2020-05-07 um 11 12 53" src="https://user-images.githubusercontent.com/3372410/81276759-c16c3180-9053-11ea-9107-5875ee592316.png">

Second it's hard to understand for people who don't know Docusaurus. None of our docs contributors has figured out how to use this correctly. This means that the time to merge for docs changes goes up and frustrates both maintainers as well as contributors.

Third it breaks previews. Assuming that - as it's documented - the import is always the first element, we get previews in Slack

<img width="631" alt="Bildschirmfoto 2020-05-07 um 11 15 52" src="https://user-images.githubusercontent.com/3372410/81277034-2162d800-9054-11ea-9c61-6280a594b333.png">

and even the HTML meta tags which is a SEO-killer:

<img width="826" alt="Bildschirmfoto 2020-05-07 um 11 16 27" src="https://user-images.githubusercontent.com/3372410/81277119-3f303d00-9054-11ea-90bd-b11fca912b2c.png">

Please consider adding this functionality to Docusaurus. I'd be happy to help!

@slorber sorry for the delay getting back to you. This is the build error I get if I use a markdown image without importing that particular image with both the following syntax:
```
![](./poisson.png)
![](poisson.png)
```
(where poisson.png is a file in the same dir as the markdown file referring to it)

```
../docs/overview/quick_start/quick_start.md
Module not found: Can't resolve 'url-loader' in '/data/sandcastle/boxes/fbsource/fbcode/beanmachine/docs/overview/quick_start'
Client bundle compiled with errors therefore further build is impossible.
../docs/overview/quick_start/quick_start.md
Module not found: Error: Can't resolve 'url-loader' in '/data/sandcastle/boxes/fbsource/fbcode/beanmachine/docs/overview/quick_start'
 @ ../docs/overview/quick_start/quick_start.md 1:5502-5610
 @ ./.docusaurus/registry.js
 @ ./node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js
 @ ./.docusaurus/routes.js
 @ ./node_modules/@docusaurus/core/lib/client/clientEntry.js
 @ multi ./node_modules/@docusaurus/core/lib/client/clientEntry.js
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

> yarn why url-loader
```
Using Yarn from /data/sandcastle/boxes/fbsource/xplat/third-party/yarn/yarn
yarn why v1.17.0-20190429.1820
[1/4] Why do we have the module "url-loader"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "url-loader@4.1.0"
info Reasons this module exists
   - "@docusaurus#core" depends on it
   - Hoisted from "@docusaurus#core#url-loader"
   - Hoisted from "@docusaurus#preset-classic#@docusaurus#plugin-content-blog#@docusaurus#mdx-loader#url-loader"
info Disk size without dependencies: "492KB"
info Disk size with unique dependencies: "596KB"
info Disk size with transitive dependencies: "3.02MB"
info Number of shared dependencies: 11
Done in 1.02s.
```

I'm using yarn 1.17.0, the one that runs as standard at facebook.

I think the `filepath:///` thing was a red herring, I found it in an issue by googling the error message, and thought it might be the new required syntax so gave it a try.

-----

So speaking for myself, I still can't get any images to work using the markdown syntax.
Maybe I'm doing something wrong, but as far as I can tell I'm following the docs.

Tried on versions 58 and 61.

To clarify, I'd be happy with the restriction to only use absolute paths instead, but they don't currently get the baseUrl prepended.

--- a/packages/docusaurus/src/webpack/base.ts
+++ b/packages/docusaurus/src/webpack/base.ts
@@ -92,6 +92,7 @@ export function createBaseConfig(
     },
     resolveLoader: {
       plugins: [PnpWebpackPlugin.moduleLoader(module)],
+      modules: ['node_modules', path.join(siteDir, 'node_modules')],
     },
     optimization: {
       removeAvailableModules: false,

